tzatziki-spring-kafka/src/main/java/com/decathlon/tzatziki/steps/KafkaSteps.java

```diff
@@ -289,38 +295,49 @@ public void that_the_current_offset_the_groupid_on_topic_is(String groupId, Stri
-                    if (offset >= 0) {
-                        consumer.seek(topicPartition, offset);
-                    } else {
-                        log.debug("offset was %s for topic %s".formatted(offset, topicPartition));
-                    }
-                });
-            }
+            containsValueInTopic(fromBeginning, topic, comparison, name, content);
         });
     }
 
+    @Then(THAT + GUARD + "(from the beginning )?the " + QUOTED_CONTENT + " topic contains" + COMPARING_WITH + " " + A + RECORD + ":$")
+    public void the_template_topic_contains(Guard guard, boolean fromBeginning, String topic, Comparison comparison, String name, String content) {
+        guard.in(objects, () -> {
+            containsValueInTopic(fromBeginning, objects.resolve(topic), comparison, name, content);
+        });
+    }
+
+    private void containsValueInTopic(boolean fromBeginning, String topic, Comparison comparison, String name, String content) {
+        Consumer<?, ?> consumer = getConsumer(name, topic);
+        List<TopicPartition> topicPartitions = awaitTopicPartitions(topic, consumer);
+        if (!consumer.assignment().containsAll(topicPartitions) || fromBeginning) {
+            consumer.assign(topicPartitions);
+            consumer.seekToBeginning(topicPartitions);
+            consumer.commitSync();
+        }
+        try {
+            ConsumerRecords<?, ?> records = consumer.poll(Duration.ofSeconds(1));
+            List<Map<String, Object>> consumerRecords = StreamSupport.stream(records.spliterator(), false)
+                    .map(record -> {
+                        Map<String, String> headers = Stream.of(record.headers().toArray())
+                                .collect(Collectors.toMap(Header::key, header -> new String(header.value())));
+                        Map<String, Object> value = Mapper.read(record.value().toString());
+                        String messageKey = record.key() != null ? String.valueOf(record.key()) : "";
+                        return Map.of("value", value, "headers", headers, "key", messageKey);
+                    })
+                    .collect(Collectors.toList());
+            comparison.compare(consumerRecords, asListOfRecordsWithHeaders(Mapper.read(objects.resolve(content))));
+        } finally {
+            TopicPartition topicPartition = new TopicPartition(topic, 0);
+            ofNullable(offsets().get(topicPartition)).ifPresent(offset -> {
+                if (offset >= 0) {
+                    consumer.seek(topicPartition, offset);
+                } else {
+                    log.debug("offset was %s for topic %s".formatted(offset, topicPartition));
+                }
+            });
+        }
+    }
+
     @Then(THAT + GUARD + "the " + VARIABLE + " topic contains (\\d+) " + RECORD + "?$")
     public void the_topic_contains_n_messages(Guard guard, String topic, int amount, String name) {
```

Comment:
In my opinion, if we introduce this templating in topic name behavior, we should also add it to all the other steps that are using topics name like this one. 
Also other like _a_message_is_consumed_from_a_topic_ ...

Reply:
yep I have hesitate